spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/RelationalEntityDeleteWriter.java

```diff
@@ -42,12 +41,15 @@
 public class RelationalEntityDeleteWriter implements EntityWriter<Object, MutableAggregateChange<?>> {
 
 	private final RelationalMappingContext context;
+	/** The previous version assigned to the instance for which a change is being written, if available */
+	@Nullable private final Number previousVersion;
```

Comment:
Can we keep the version number in `MutableAggregateChange` instead of moving it around? The changed constructor below creates an incompatible API change.

Reply:
Could you please provide some more context on how this creates an incompatible API change? It isn't immediately obvious to me. Would this apply to the same changes made to the [`RelationalEntityUpdateWriter`](https://github.com/spring-projects/spring-data-relational/pull/1211/files#diff-2ee8eaa149439fd40b96bccae27460ee4b66270d5b63b5970ba5fb8445e88cff) as well?

Reply:
The constructor `RelationalEntityDeleteWriter(RelationalMappingContext context)` has been changed to `RelationalEntityDeleteWriter(RelationalMappingContext context, @Nullable Number previousVersion)` and every place that wants to instantiate `RelationalEntityDeleteWriter` is required to use a different constructor signature.

Other `Relational…Writer`s are affected in the same way.

Related, `MutableAggregateChange` has been defining `getPreviousVersion()` and now the method was removed from the interface which is also a breaking change.

Maybe we can get together to revisit and discuss the further approach directly.

Reply:
Ok I will remove the commit that introduces these changes to version number in `MutableAggregateChange`. It turns out that these changes are quite unrelated to the goal of this pull request and do not belong here.